server/stream.go

```diff
@@ -168,7 +177,7 @@ func (a *Account) AddStreamWithStore(config *StreamConfig, fsConfig *FileStoreCo
 
 	// Setup the internal client.
 	c := s.createInternalJetStreamClient()
-	mset := &Stream{jsa: jsa, config: cfg, client: c, consumers: make(map[string]*Consumer)}
+	mset := &Stream{jsa: jsa, config: cfg, client: c, consumers: make(map[string]*Consumer), nosubj: noSubjectsOK}
```

Comment:
Let's move to stream and consumer config as internal fields in config struct.

`interestNotRequired`

Reply:
What about the issue with persisting the config? I will experiment to see if a non exported config option works, but I believe that it would not be persisted..

Reply:
Good point, this just feels weird passing around a simple bool to the internal functions etc.

Reply:
I really would want to make those configuration options, and not have to carry over some boolean in the function calls, but we have the problem of persistence. I can overcome that with implementing MarshalJSON/UnmarshalJSON for FileStreamInfo and FileConsumerInfo if you are ok with that approach?

Reply:
Yes I agree.